sklearn/decomposition/tests/test_kernel_pca.py

```diff
@@ -79,7 +79,10 @@ def test_kernel_pca_invalid_parameters():
     ValueError.
     """
     with pytest.raises(ValueError):
-        KernelPCA(10, fit_inverse_transform=True, kernel="precomputed")
+        estimator = KernelPCA(
+            n_components=10, fit_inverse_transform=True, kernel="precomputed"
+        )
+        estimator.fit(np.random.randn(10, 10))
```

Comment:
This test:
 - has to be modified to make sure the warning is only thrown when `fit` is called,
 - can better be scoped by adding a check for the `ValueError` message.
 
I can't suggest, but this can be done :

```python
    estimator = KernelPCA(
        n_components=10, fit_inverse_transform=True, kernel="precomputed"
    )
    err_ms = "Cannot fit_inverse_transform with a precomputed kernel"
    with pytest.raises(ValueError, match=err_ms):
        estimator.fit(np.random.randn(10, 10))
```